addons/website_sale/models/sale_order.py

```diff
@@ -335,8 +335,9 @@ def _cart_find_product_line(
         :rtype: `sale.order.line` recordset
         """
         self.ensure_one()
+        product = self.env['product.product'].browse(product_id)
 
-        if not self.order_line:
+        if not self.order_line or product.type == 'combo':
```

Comment:
Could be kept in another if to avoid browsing the product when the cart is empty.
Not critical ofc, feel free to keep it like this if you prefer.